### ISSUE TITLE
Edit Add Units Bug 1.9.0.0.2104

### ISSUE THREAD

**Issue by Cernelius** [triplea-game/triplea#831]
When you click on Game/Enable Edit Mode/Add Units, you may receive pointless console errors, in TripleA 1.9.0.0.2104.

The Edit behaviour appears to be the same as the current release (allowing editing all units the player has images for, either in the homonymous subfolder inside the units folder of the map folder or in assets), but with the console errors now added, in all cases a player lacks images for any units (also the ones not belonging to its own frontier).

This should be solved either by:
1) keeping the current release behaviour and avoiding the mentioned errors to be logged;
2) keeping the current release behaviour but keeping logging the mentioned errors, but avoiding the console popping up in that instance.

The reason for keeping the current errors logging is that I tend to believe it would advisable mapmakers provide images of each unit to all players, also the ones supposed never to have them (for example, I believe that all players of World At War should have a Marine image, also those supposed to never have Marine (I'm not going to do it)).

I've released "World At War" 2.0 here:
http://tripleadev.1671093.n2.nabble.com/WORLD-AT-WAR-tp5862407p7592327.html

I have the following console errors in World At War 2.0, upon clicking on Game/Enable Edit Mode/Add Units, during Germans turn 1:

> triplea.engine.version.bin:1.9
> MISSING IMAGE (this unit or image will be invisible): B.Transport  looking in: units/Germans/B.Transport.png
> MISSING IMAGE (this unit or image will be invisible): S.Bomber  looking in: units/Germans/S.Bomber.png
> MISSING IMAGE (this unit or image will be invisible): Mech.Rocket  looking in: units/Germans/Mech.Rocket.png
> MISSING IMAGE (this unit or image will be invisible): B.Carrier  looking in: units/Germans/B.Carrier.png
> MISSING IMAGE (this unit or image will be invisible): L.Bomber  looking in: units/Germans/L.Bomber.png
> MISSING IMAGE (this unit or image will be invisible): L.Fighter  looking in: units/Germans/L.Fighter.png

To sum it up, I tend to believe this errors logging may be a good addition for mapmakers, and I advice it to be kept, but surely no pop up warning should be automatically given the user, upon editing.

**Comment by DanVanAtta** [comment]
Is this only a problem for World at War?
Looks related to an update we made to not crash the game when a unit image is missing. With that fix it was intended that map makers should know right away, and get a fix in before the map is released. This implies the previous behavior would have been a unit not found error, and then map rendering would stop and the game would be unplayable. Is that consistent with what you are seeing?

**Comment by DanVanAtta** [comment]
The pop-up will appear whenever the game tries to load an image that does not exist. We need to verify that it is correctly appearing when it does.  The decision for it to appear as a on-the-top pop-up was explicit. 

@Cernelius , please attach a link to the latest version of a map that has this problem. Please also mention if any of the existing 1.9.0.0 maps have this problem.

**Comment by Cernelius** [comment]
Yeah, this is what happens when a developer (Veqryn) decides about behaviours without making a documentation anywhere (I guess) about the choices and some other developers takes over without the previous developer being even around to tell what's what.

So:

- The edit mode behaviour is and was that all units the player has images for are editable (Veqryn's decision, I guess).

- While it should be Veqryn telling you this, I surmise you are wrong to assume that whenever the game tries to load an image that does not exist, then this is a map bug and it was intended to never happen (edit is not the only case), or at least I'm not aware it was ever requested or even suggested the mapmakers to provide all players with one image for each unit in the game.

- If you want users to receive an error pop-up every single time they add units with edit in most maps with different units per player, then everything is working just fine.

- Adding units with edit appears to be working fine, or at least the same way as in the past, aside only from receiving the aforementioned error pop-up.

You can try with "Middle Earth", editing into some units for "Saruman":

> triplea.engine.version.bin:1.9
> MISSING IMAGE (this unit or image will be invisible): towerGuard  looking in: units/Saruman/towerGuard.png
> MISSING IMAGE (this unit or image will be invisible): hunters  looking in: units/Saruman/hunters.png
> MISSING IMAGE (this unit or image will be invisible): lightCavalry  looking in: units/Saruman/lightCavalry.png
> MISSING IMAGE (this unit or image will be invisible): cannon  looking in: units/Saruman/cannon.png
> MISSING IMAGE (this unit or image will be invisible): oliphant  looking in: units/Saruman/oliphant.png
> MISSING IMAGE (this unit or image will be invisible): cutter  looking in: units/Saruman/cutter.png
> MISSING IMAGE (this unit or image will be invisible): axeInfantry  looking in: units/Saruman/axeInfantry.png
> MISSING IMAGE (this unit or image will be invisible): greatship  looking in: units/Saruman/greatship.png
> MISSING IMAGE (this unit or image will be invisible): shooters  looking in: units/Saruman/shooters.png
> MISSING IMAGE (this unit or image will be invisible): raiderCavalry  looking in: units/Saruman/raiderCavalry.png
> MISSING IMAGE (this unit or image will be invisible): eagles  looking in: units/Saruman/eagles.png
> MISSING IMAGE (this unit or image will be invisible): swanBoat  looking in: units/Saruman/swanBoat.png
> MISSING IMAGE (this unit or image will be invisible): stabbers  looking in: units/Saruman/stabbers.png
> MISSING IMAGE (this unit or image will be invisible): undead  looking in: units/Saruman/undead.png
> MISSING IMAGE (this unit or image will be invisible): warWagon  looking in: units/Saruman/warWagon.png
> MISSING IMAGE (this unit or image will be invisible): clipper  looking in: units/Saruman/clipper.png
> MISSING IMAGE (this unit or image will be invisible): barge  looking in: units/Saruman/barge.png
> MISSING IMAGE (this unit or image will be invisible): chariotArchers  looking in: units/Saruman/chariotArchers.png
> MISSING IMAGE (this unit or image will be invisible): trebuche  looking in: units/Saruman/trebuche.png
> MISSING IMAGE (this unit or image will be invisible): galleon  looking in: units/Saruman/galleon.png
> MISSING IMAGE (this unit or image will be invisible): xbowInfantry  looking in: units/Saruman/xbowInfantry.png
> MISSING IMAGE (this unit or image will be invisible): zepplin  looking in: units/Saruman/zepplin.png
> MISSING IMAGE (this unit or image will be invisible): archers  looking in: units/Saruman/archers.png
> MISSING IMAGE (this unit or image will be invisible): berzerkers  looking in: units/Saruman/berzerkers.png
> MISSING IMAGE (this unit or image will be invisible): hobbits  looking in: units/Saruman/hobbits.png
> MISSING IMAGE (this unit or image will be invisible): mortar  looking in: units/Saruman/mortar.png
> MISSING IMAGE (this unit or image will be invisible): ents  looking in: units/Saruman/ents.png
> MISSING IMAGE (this unit or image will be invisible): bats  looking in: units/Saruman/bats.png
> MISSING IMAGE (this unit or image will be invisible): longship  looking in: units/Saruman/longship.png
> MISSING IMAGE (this unit or image will be invisible): phalanx  looking in: units/Saruman/phalanx.png
> MISSING IMAGE (this unit or image will be invisible): catapult  looking in: units/Saruman/catapult.png
> MISSING IMAGE (this unit or image will be invisible): balrog  looking in: units/Saruman/balrog.png
> MISSING IMAGE (this unit or image will be invisible): griffin  looking in: units/Saruman/griffin.png
> MISSING IMAGE (this unit or image will be invisible): heavyCavalry  looking in: units/Saruman/heavyCavalry.png
> MISSING IMAGE (this unit or image will be invisible): corsairs  looking in: units/Saruman/corsairs.png
> MISSING IMAGE (this unit or image will be invisible): warChariot  looking in: units/Saruman/warChariot.png
> MISSING IMAGE (this unit or image will be invisible): wargs  looking in: units/Saruman/wargs.png
> MISSING IMAGE (this unit or image will be invisible): nazgul  looking in: units/Saruman/nazgul.png
> MISSING IMAGE (this unit or image will be invisible): caravel  looking in: units/Saruman/caravel.png
> MISSING IMAGE (this unit or image will be invisible): warWolves  looking in: units/Saruman/warWolves.png
> MISSING IMAGE (this unit or image will be invisible): spearInfantry  looking in: units/Saruman/spearInfantry.png

**Comment by DanVanAtta** [comment]
@Cernelius redundant information between issues is not helpful. This last message was already posted, repeating it here is not helpful. 

Closing to discourage the mirror conversation, and it is redundant to #830
